lib/Parser/Parse.cpp

```diff
@@ -10473,8 +10473,16 @@ void Parser::FinishDeferredFunction(ParseNodePtr pnodeScopeList)
             pnodeFnc->sxFnc.pnodeVars = nullptr;
             m_ppnodeVar = &pnodeFnc->sxFnc.pnodeVars;
 
+            ParseNodePtr pnodeFncSaveNonLambda = nullptr;
+
+            pnodeFncSaveNonLambda = m_currentNodeNonLambdaFunc;
```

Comment:
`GetCurrentFunctionNode`() should not be null. In any case there are better ways to check if the current function you are parsing is the global function.